### PR TITLE
v2.1.0-dev

### PR DESCRIPTION
**Pull request by nkallima**
Time to merge 2.1.0-dev to master branch?

### CODE REVIEW

File: benchmarks/activesetbench.c [hunk 1]
```diff
@@ -55,7 +55,7 @@ int main(int argc, char *argv[]) {
     d2 = getTimeMillis();
 
     printf("time: %d (ms)\tthroughput: %.2f (millions ops/sec)\t", (int)(d2 - d1), bench_args.runs * bench_args.nthreads / (1000.0 * (d2 - d1)));
-    printStats(bench_args.nthreads);
+    printStats(bench_args.nthreads, bench_args.total_runs);
```
Comment (Spiros-N-Agathos) [inline benchmarks/activesetbench.c:58]: Here (and in all files) you could just pass a const pointer to BenchArgs struct, also a way to make future proof.

File: libconcurrent/primitives/primitives.c [hunk 2]
```diff
@@ -135,10 +135,12 @@ inline void freeMemory(void *ptr, size_t size) {
 }
 
 inline int64_t getTimeMillis(void) {
-    struct timeb tm;
+    struct timespec tm;
 
-    ftime(&tm);
-    return 1000 * tm.time + tm.millitm;
+    if (clock_gettime(CLOCK_MONOTONIC, &tm) == -1) {
```
Comment (Spiros-N-Agathos) [inline libconcurrent/primitives/primitives.c:140]: There are some issues in WSL:
https://github.com/Microsoft/WSL/issues/3840
but i can live with that.

**Comment by Spiros-N-Agathos** [comment]
Trying to review the changes I saw that 2 files had execution rights, check the attached image:
Screenshot from 2020-12-29 16-24-14

File: libconcurrent/primitives/primitives.c (hunk 2 above)
Comment (nkallima) [reply libconcurrent/primitives/primitives.c:140]: I tested it in WSL2 and it seems to work correctly. Dis you test it with WSL1?

File: benchmarks/activesetbench.c (hunk 1 above)
Comment (nkallima) [reply benchmarks/activesetbench.c:58]: The functionality provided by stats.c and stats.h is provided by the runtime and I think should not be strongly depended on BenchArgs, which is functionality used only by the benchmarks and not the runtime. However, we may re-consider it in the next release.

Comment (nkallima) [reply benchmarks/activesetbench.c:58]: I tested in OpenSUSE 15.2 over windows 20H2 + wsl1 and it works fine. Thus, I think check this point as resolved.

File: libconcurrent/primitives/primitives.c (hunk 2 above)
Comment (nkallima) [reply libconcurrent/primitives/primitives.c:140]: I tested in OpenSUSE 15.2 over windows 20H2 + wsl1 and it works fine. Thus, I mark this point as resolved.

**Comment by nkallima** [comment]
> Trying to review the changes I saw that 2 files had execution rights, check the attached image:
> Screenshot from 2020-12-29 16-24-14

Done!

File: benchmarks/activesetbench.c (hunk 1 above)
Comment (Spiros-N-Agathos) [reply benchmarks/activesetbench.c:58]: Fine by me

**Pull request merged**